AK/FileSystemPath.cpp

```diff
@@ -37,6 +37,11 @@ void FileSystemPath::canonicalize()
     }
 
     m_basename = canonical_parts.last();
+    auto name_parts = m_basename.split('.');
+    m_title = name_parts[0].characters();
```

Comment:
No need for `.characters()` since that is both overly verbose and will cause a `String` copy.